admin/src/js/modules/ajax-download.js

```diff
@@ -4,16 +4,27 @@
 
   const DEFAULT_FILE_NAME = 'download';
 
+  const getFileName = (headers) => {
+    const contentDisposition = headers.get('content-disposition');
+    const match = /filename=(?:"|)(.*?)(?:"|;|$)/.exec(contentDisposition);
+    return match ? match[1].trim() : DEFAULT_FILE_NAME;
+  };
+
   /**
    * Prompts the user to download a file given a url.
    */
   exports.download = function(url) {
-    const element = document.createElement('a');
-    element.setAttribute('href', url);
-    element.setAttribute('download', DEFAULT_FILE_NAME);
-    element.style.display = 'none';
-    document.body.appendChild(element);
-    element.click();
-    document.body.removeChild(element);
+    return fetch(url)
+      .then(response => response.blob()
+        .then(blob => {
+          const blobUrl = URL.createObjectURL(blob);
+          const link = document.createElement('a');
+          link.style.display = 'none';
+          link.href = blobUrl;
+          link.setAttribute('download', getFileName(response.headers));
+          document.body.appendChild(link);
+          link.click();
+          URL.revokeObjectURL(blobUrl);
+        }));
```

Comment:
Minor
Maybe I was reading too fast, but my mind automatically closed the first `then` at the end of `response.blob()` and thought the second `then` had bad indentation. This might help readability a tiny bit more while keeping code flat:
```suggestion
      .then(response => response.blob())
      .then(blob => {
        const blobUrl = URL.createObjectURL(blob);
        const link = document.createElement('a');
        link.style.display = 'none';
        link.href = blobUrl;
        link.setAttribute('download', getFileName(response.headers));
        document.body.appendChild(link);
        link.click();
        URL.revokeObjectURL(blobUrl);
      }));
```